internal/pkg/convert/node.go

```diff
@@ -327,8 +348,22 @@ func (node *NodeLLB) finalize(root llb.State) llb.State {
 }
 
 // Build converts PackageNode to buildkit LLB.
-func (node *NodeLLB) Build() (llb.State, error) {
-	if state, ok := node.Graph.cache[node.PackageNode]; ok {
+func (node *NodeLLB) Build(platform string) (llb.State, error) {
+	cacheSt := cacheState{
+		PackageNode: node.PackageNode,
+	}
+
+	if platform != "" {
+		v1Platform, err := node.convertPlatform(platform)
+		if err != nil {
+			return llb.Scratch(), err
+		}
+
+		node.Platform = Platform{v1Platform}
```

Comment:
this looks strange that it sets it on the node, does it make more sense to pass platform to `NewNodeLLB` ?

Reply:
fixed